pkg/registry/apis/identity/legacy/queries_test.go

```diff
@@ -0,0 +1,207 @@
+package legacy
```

Comment:
I feel like these kinds of test is likely to be duplicated across different apis. Could be nice to create a more reusable test suit that all can reuse

Reply:
i'll tackle that in the next PR -- this one is already kinda big :)